src/main/java/org/openstreetmap/atlas/checks/validation/areas/ShadowDetectionCheck.java

```diff
@@ -104,8 +105,8 @@ public boolean validCheckForObject(final AtlasObject object)
     protected Optional<CheckFlag> flag(final AtlasObject object)
     {
         // Gather connected building parts and check for a connection to the ground
-        final Set<AtlasObject> floatingParts = this.getFloatingParts(object);
-        if (!floatingParts.isEmpty())
+        final Tuple<Set<AtlasObject>, Boolean> connectedParts = this.getConnectedParts(object);
+        if (connectedParts.getSecond())
```

Comment:
@danielduhh, the code smell is because `connectedParts.getSecond()` is of type `Boolean` instead of `boolean`. The resolution seems to be to cast it to a `boolean` which seems a bit odd/redundant. Sonar is just worried about it being null, which can't happen in this case.